packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -127,3 +148,51 @@ class _ProxyOpacityProvider<T extends Object> implements OpacityProvider {
   @override
   set opacity(double value) => target.setOpacity(value, paintId: paintId);
 }
+
+class _MultiPaintOpacityProvider<T extends Object> implements OpacityProvider {
+  _MultiPaintOpacityProvider(this.paintIds, this.target) {
+    final maxOpacity = opacity;
+
+    _opacityRatios = (paintIds != null)
+        ? List<double>.generate(
+            paintIds!.length,
+            (index) =>
+                target.getOpacity(paintId: paintIds!.elementAt(index)) /
+                maxOpacity,
+          )
+        : null;
+  }
+
+  final List<T>? paintIds;
+  final HasPaint<T> target;
+  late final List<double>? _opacityRatios;
+
+  @override
+  double get opacity {
+    var maxOpacity = 0.0;
```

Comment:
```suggestion
    var maxOpacity = target.getOpacity();
```

Reply:
That results in incorrect output. `getOpacity()` returns the opacity of `paint` which is not part of the `_paints` map. So if users provide a list of `paintIds`, `maxOpacity` will first get initialised with the default value of 1 from `paint`s opacity. As a result the result of `max` inside the loop will always be 1.